osf/models/node.py

```diff
@@ -1351,7 +1354,7 @@ def add_contributor_registered_or_not(self, auth, user_id=None,
                                  permissions=permissions, send_email=send_email, save=True)
         else:
             contributor = get_user(email=email)
-            if contributor:
+            if contributor and contributor.is_registered:
```

Comment:
Also, can you confirm that adding the same unreg contrib twice doesn't cause problems? I think it will just overwrite the unclaimed record, but might be worth checking.